src/components/Activity.jsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { useState, useEffect } from 'react';
+import {STATIC_DATABASE_URL} from '../config.json';
+import axios from 'axios';
+import { useParams } from 'react-router-dom';
+import { FaBicycle, FaRunning, FaWalking, FaClock, FaMapMarkerAlt, FaLeaf, FaPiggyBank, FaMap } from 'react-icons/fa';
+
+const Activity = () => {
+    const [activities, setActivities] = useState([]);
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState(null);
+
+    const activityIcons = {
+        cycling: <FaBicycle className="text-blue-500 text-4xl" />,
+        running: <FaRunning className="text-red-500 text-4xl" />,
+        walking: <FaWalking className="text-green-500 text-4xl" />,
+    };
+  
+    useEffect(() => {
+      const fetchActivities = async () => {
+        try {
+          const response = await axios.get(
+            `${STATIC_DATABASE_URL}/activities`,
+          );
+          if (response.status != 200) {
+            throw new Error('Network response was not ok');
+          }
+          setActivities(response.data); 
+          setLoading(false);
+        } catch (error) {
+          setError(error);
+          setLoading(false);
+        }
+      };
+  
+      fetchActivities();
+    }, []);
+
+    if (loading) {
+        return <div className="max-w-4xl mx-auto p-4">Loading...</div>;
+    }
+
+    if (error) {
+        return <div className="max-w-4xl mx-auto p-4">Error: {error.message}</div>;
+    }
+
+    const { id } = useParams();
+    const activity = activities.find(act => act.id === parseInt(id));
```

Comment:
This line is failing on my machine as `act.id` is a string, and `parseInt(id)` is an int. It worked once I removed `parseInt`. (although will change once we use the backend). Jsonserver is also quite nice so you can do something like:
```
const response = await axios.get(`${STATIC_DATABASE_URL}/activities?id=${id}`);
```